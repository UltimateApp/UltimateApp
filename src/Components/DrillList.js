import React from 'react';
import { StyleSheet, View, Text, ImageBackground, TouchableOpacity, FlatList } from 'react-native';

import * as list from '../styles/list.style';
import { DrillTypes } from '../Fixtures';

const DrillList = props => {
  const drillsToDisplay = props.drillsToDisplay;
  const navigation = props.navigation;

  const renderDrill = ({ item }) => {
    const { title, type, source, image, goals } = item;

    const imageMainData = type === DrillTypes.TECHNICAL ? 'minimalPlayersNumber' : 'durationInMinutes';
    const imageMainDataLegend = type === DrillTypes.TECHNICAL ? 'players' : 'min.';
    return (
      <TouchableOpacity style={list.item} onPress={() => navigation.navigate('DrillPage', { drill: item })}>
        <ImageBackground source={{ uri: image }} style={list.image} imageStyle={list.imageOpacity}>
          <Text style={{ ...list.imageText, ...list.imageTextMain }}>{item[imageMainData]}+</Text>
          <Text style={list.imageText}>{imageMainDataLegend}</Text>
        </ImageBackground>
        <View style={list.contentContainer}>
          <Text style={list.source}>{source}</Text>
          <Text style={list.title}>{title}</Text>
          <Text style={list.numberOfPlayers}>{goals.join(', ')}</Text>
        </View>
      </TouchableOpacity>
    );
  };

  return (
    <View>
      <FlatList data={drillsToDisplay} keyExtractor={item => item.id.toString()} renderItem={renderDrill} />
    </View>
  );
};

export default DrillList;
