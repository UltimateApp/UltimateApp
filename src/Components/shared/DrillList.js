import React from 'react';
import { View, Text, ImageBackground, TouchableOpacity, FlatList } from 'react-native';

import * as list from '../../styles/list.style';
import { DrillTypes } from '../../Fixtures';

const DrillList = props => {
  const { navigation, drillsToDisplay } = props;
  const onDrillPress = props.onDrillPress || (item => navigation.navigate('DrillPage', { drill: item }));

  const renderDrill = ({ item }) => {
    const { title, type, source, image, goals } = item;

    const imageMainData = type === DrillTypes.FRISBEE ? 'minimalPlayersNumber' : 'durationInMinutes';
    const imageMainDataLegend = type === DrillTypes.FRISBEE ? 'players' : 'min';
    return (
      <TouchableOpacity style={[list.item, props.ItemComponentStyle]} onPress={() => onDrillPress(item)}>
        <ImageBackground source={{ uri: image }} style={list.image} imageStyle={list.imageOpacity}>
          <Text style={{ ...list.imageText, ...list.imageTextMain }}>{item[imageMainData]}</Text>
          <Text style={list.imageText}>{imageMainDataLegend}</Text>
        </ImageBackground>
        <View style={list.itemContentContainer}>
          <Text style={list.source}>{source}</Text>
          <Text style={list.title}>{title}</Text>
          <Text style={list.numberOfPlayers}>{goals.join(', ')}</Text>
        </View>
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      {...props}
      data={drillsToDisplay}
      keyExtractor={item => item.id.toString()}
      renderItem={renderDrill}
      contentContainerStyle={list.contentContainer}
    />
  );
};

export default DrillList;
