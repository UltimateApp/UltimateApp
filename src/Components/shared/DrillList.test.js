import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import { createDrill } from '../../Fixtures/TestFixtures';

import DrillList from './DrillList';

describe('<DrillList />', () => {
  it('renders correctly', () => {
    const drills = [createDrill()];
    const tree = renderer.create(<DrillList drillsToDisplay={drills} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
