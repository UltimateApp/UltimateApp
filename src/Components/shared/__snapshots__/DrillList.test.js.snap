// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillList /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 40,
    }
  }
  data={
    Array [
      Object {
        "author": "Author",
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 1,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "illustrationSource": "462695757",
            "illustrationType": "vimeo",
            "subtitle": "This is a subtitle",
            "title": "Warmup",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
      },
    ]
  }
  disableVirtualization={false}
  drillsToDisplay={
    Array [
      Object {
        "author": "Author",
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 1,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "illustrationSource": "462695757",
            "illustrationType": "vimeo",
            "subtitle": "This is a subtitle",
            "title": "Warmup",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "height": 80,
            "marginBottom": 20,
            "opacity": 1,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0,0,0)",
              "borderRadius": 5,
              "height": 80,
              "justifyContent": "center",
              "marginRight": 10,
              "padding": 10,
              "width": 80,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 5,
                  "opacity": 0.5,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            10
          </Text>
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 14,
                "textAlign": "center",
              }
            }
          >
            min
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 5,
              "paddingBottom": 10,
              "paddingRight": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Author
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "flex": 3,
                "flexWrap": "wrap",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Fitness Drill Title
          </Text>
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Full body
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
